internal/p2p/pex/reactor.go

```diff
@@ -181,13 +184,27 @@ func (r *Reactor) processPexCh(ctx context.Context, pexCh *p2p.Channel) {
 			return
 
 		case <-timer.C:
+			// back off sending peer requests if there's none available.
+			// Let the loop continue to handle incoming pex messages
+			if noAvailablePeerFailCounter > 0 {
+				waitPeriod := float64(noAvailablePeersWaitPeriod) * float64(noAvailablePeerFailCounter)
+				if time.Since(lastNoAvailablePeersTime).Seconds() < time.Duration(waitPeriod).Seconds() {
+					r.logger.Debug(fmt.Sprintf("waiting for more peers to become available still in the waitPeriod=%f\n", time.Duration(waitPeriod).Seconds()))
+					continue
+				}
+			}
+
 			// Send a request for more peer addresses.
 			if err := r.sendRequestForPeers(ctx, pexCh); err != nil {
+				r.logger.Error("failed to send request for peers", "err", err)
+				if strings.Contains(err.Error(), "no available peers") {
```

Comment:
Instead of string pattern matching, a better or more elegant way might be to define a new Error instance NoAvailablePeersError and if the returned error is that then we increment the counter

Reply:
good point!